doc/glossary.rst

```diff
@@ -918,6 +918,19 @@ Class APIs and Estimator Types
         outliers have score below 0.  :term:`score_samples` may provide an
         unnormalized score per sample.
 
+    outlier rejector
```

Comment:
I think we should remove this and just add a note under `outlier detector` that if `fit_resample` is provided, it should act as an outlier rejector, returning the training data with outliers removed.